feature-libs/organization/administration/core/store/effects/b2b-user.effect.ts

```diff
@@ -105,6 +108,51 @@ export class B2BUserEffects {
     )
   );
 
+  @Effect()
+  createB2BUserAndAssignToApprovers$: Observable<
+    | B2BUserActions.CreateB2BUserSuccess
+    | OrgUnitActions.AssignApprover
+    | B2BUserActions.CreateB2BUserFail
+    | OrganizationActions.OrganizationClearData
+  > = this.actions$.pipe(
+    ofType(B2BUserActions.CREATE_B2B_USER_AND_ASSIGN_TO_APPROVERS),
+    map(
+      (action: B2BUserActions.CreateB2BUserAndAssignToApprovers) =>
+        action.payload
+    ),
+    switchMap((payload) =>
+      this.b2bUserConnector.create(payload.userId, payload.orgCustomer).pipe(
+        concatMap((data) => {
+          const assignApproverPayload = {
+            userId: payload.userId,
+            orgUnitId: payload.orgCustomer.orgUnit.uid,
+            orgCustomerId: data.customerId,
+            roleId: 'b2bapprovergroup',
+          };
+
+          this.routingService.go({
+            cxRoute: 'userDetails',
+            params: { customerId: data.customerId },
+          });
```

Comment:
Please check what we have done here with createB2BUser$. 
Both scenarios should redirect to other places: 
- user list / add -> user details
- unit / users / add -> unit / users